apps/signal/56-forms-and-signal/src/app/order.component.ts

```diff
@@ -55,11 +57,18 @@ import { products } from './products';
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export default class OrderComponent {
+  private route = inject(ActivatedRoute);
+
+  query$ = this.route.queryParams.pipe(map((q) => q['quantity'] as number));
```

Comment:
you can use the snapshot version since you only want it when the component loads